packages/modular-scripts/src/cli.ts

```diff
@@ -227,27 +230,85 @@ function addApp(name: string) {
 function test(args: string[]) {
   const modularRoot = getModularRoot();
 
-  const appDirent = fs
-    .readdirSync(path.join(modularRoot, 'packages'), {
-      withFileTypes: true,
-    })
-    .find(
-      (p) =>
-        p.isDirectory() &&
-        isModularType(path.join(modularRoot, 'packages', p.name), 'app'),
+  let argv = mri(process.argv.slice(3))._.concat([
+    '--config',
+    path.join(__dirname, '..', 'jest-config.js'),
+  ]);
+
+  // Watch unless on CI or explicitly running all tests
+  if (!process.env.CI && args.indexOf('--watchAll=false') === -1) {
+    // https://github.com/facebook/create-react-app/issues/5210
+    argv.push('--watchAll');
+  }
+
+  // via https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/test.js
+
+  // This is a very dirty workaround for https://github.com/facebook/jest/issues/5913.
+  // We're trying to resolve the environment ourselves because Jest does it incorrectly.
+  // TODO: remove this as soon as it's fixed in Jest.
+
+  function resolveJestDefaultEnvironment(name: string) {
+    const jestDir = path.dirname(
+      resolve.sync('jest', {
+        basedir: __dirname,
+      }),
+    );
+    const jestCLIDir = path.dirname(
+      resolve.sync('jest-cli', {
+        basedir: jestDir,
+      }),
     );
-  if (!appDirent) {
-    // TODO: this is a ridiculous requirement and we should fix it
-    throw new Error(
-      'modular cannot run tests without at least one app. Add an app and try again.',
+    const jestConfigDir = path.dirname(
+      resolve.sync('jest-config', {
+        basedir: jestCLIDir,
+      }),
     );
+    return resolve.sync(name, {
+      basedir: jestConfigDir,
+    });
   }
+  const cleanArgv = [];
+  let env = 'jsdom';
+  let next;
+  do {
+    next = argv.shift();
+    if (next === '--env') {
+      env = argv.shift() as string;
+    } else if (next?.indexOf('--env=') === 0) {
+      env = next.substring('--env='.length);
+    } else {
+      cleanArgv.push(next);
+    }
+  } while (argv.length > 0);
+  // @ts-ignore
+  argv = cleanArgv;
+  let resolvedEnv;
+  try {
+    resolvedEnv = resolveJestDefaultEnvironment(`jest-environment-${env}`);
+  } catch (e) {
+    // ignore
+  }
+  if (!resolvedEnv) {
+    try {
+      resolvedEnv = resolveJestDefaultEnvironment(env);
+    } catch (e) {
+      // ignore
+    }
+  }
+  const testEnvironment = resolvedEnv || env;
+  argv.push('--env', testEnvironment || '');
+
+  // ends the section copied from CRA
 
-  return execSync(cracoBin, ['test', '--config', cracoConfig, ...args], {
-    cwd: path.join(modularRoot, 'packages', appDirent.name),
+  return execSync(jestBin, argv, {
+    cwd: modularRoot,
     log: false,
+    cleanup: true,
```

Comment:
This is already implicit from above, no? 

https://github.com/jpmorganchase/modular/pull/155/files#diff-228bd4df6d9d99fd05a68843f3ecd0c638ee9a58849130167f7ddd25a39c6cd2R44

Reply:
yeah, will remove